src/d3d9/d3d9_swapchain.cpp

```diff
@@ -156,10 +162,49 @@ namespace dxvk {
       return D3D_OK;
     } catch (const DxvkError& e) {
       Logger::err(e.message());
+#ifdef _WIN32
+      return BlitGDI(window);
+#else
       return D3DERR_DEVICEREMOVED;
+#endif
     }
   }
 
+#ifdef _WIN32
+  HRESULT D3D9SwapChainEx::BlitGDI(HWND Window) {
+    if (!m_warnedAboutFallback || true) {
+      Logger::warn("Using GDI for swapchain presentation. This will impact performance.");
+      m_warnedAboutFallback = true;
+    }
```

Comment:
DXVK uses 
```cpp
    static bool s_errorShown = false;

    if (!std::exchange(s_errorShown, true))
      Logger::warn(...);
```
pretty much everywhere. Or I missed something and this case is special?

Reply:
You didn't miss anything. Good point.